### PR TITLE
Feature display density

### PR DESCRIPTION
**Pull request by devendra940**
This will work as below,
There will be a dropdown, and on the selection of drop-down options, it will apply display comfortable and display compact view to the table.
By default there will be a Comfortable option selected in the drop-down menu and on the selection of compact, it will apply a compact view to the table.

Display-Density : - 1) Comfortable (Default Selected)
                                 2) Compact

### CODE REVIEW

File: src/app/table/table-footer/display-density/display-density.component.scss [hunk 1]
```diff
@@ -0,0 +1,17 @@
+@import "../../../../styles/_global-variables";
```
Comment (APJohns) [inline src/app/table/table-footer/display-density/display-density.component.scss]: Revert styling, please focus on functionality.

File: src/app/table/table-footer/table-footer.component.scss [hunk 2]
```diff
@@ -0,0 +1,10 @@
+@import "../../../styles/_global-variables";
```
Comment (APJohns) [inline src/app/table/table-footer/table-footer.component.scss]: Revert styling, please focus on functionality

File: src/app/table/table-footer/pagination/pagination.component.scss [hunk 3]
```diff
@@ -2,12 +2,7 @@
 
 .table-pagination
```
Comment (APJohns) [inline src/app/table/table-footer/pagination/pagination.component.scss]: Revert styling, please focus on functionality

File: src/app/table/table.component.ts [hunk 4]
```diff
@@ -121,6 +122,19 @@ export class TableComponent implements OnInit, OnDestroy {
     }
   }
 
+  setDisplayDensityValue(densityName: { target: { value: string; }; }) {
```
Comment (APJohns) [inline src/app/table/table.component.ts]: Rename this to *setDisplayDensity*

**Comment by APJohns** [comment]
Looks like the we are missing the persistence functionality. (See bullet 2 in Acceptance https://epsilonjira.atlassian.net/browse/AUB-4659). I'd recommend looking into utilizing localStorage to accomplish this.

**Comment by APJohns** [comment]
Master is too far ahead to resolve these conflicts so we had to decline this PR.

File: src/app/table/table.component.ts [hunk 5]
```diff
@@ -121,6 +122,19 @@ export class TableComponent implements OnInit, OnDestroy {
...
+    if (densityName === undefined) { return null; }
+    if (densityName.target.value === 'Comfortable') {
+      this.densityClassName = null;
+    } else {
+      this.densityClassName = 'table-compact';
+    }
+  }
+
+  getDisplayDensityValue() {
```
[the remark this answers is not in the document]
Comment (devendra940) [reply src/app/table/table.component.ts]: @APJohns returning just densityClassName

File: src/app/table/table.component.ts (hunk 4 above)
Comment (devendra940) [reply src/app/table/table.component.ts]: @APJohns  Fixed

File: src/app/table/table.component.html [hunk 6]
```diff
@@ -1,4 +1,4 @@
-<table class="table">
+<table class="table" [ngClass]="getDisplayDensityValue()">
```
[the remark this answers is not in the document]
Comment (devendra940) [reply src/app/table/table.component.html]: @APJohns  removed function only returning class name

File: src/app/table/table-footer/table-footer.component.scss (hunk 2 above)
Comment (devendra940) [reply src/app/table/table-footer/table-footer.component.scss]: @APJohns  removed CSS

File: src/app/table/table-footer/pagination/pagination.component.scss (hunk 3 above)
Comment (devendra940) [reply src/app/table/table-footer/pagination/pagination.component.scss]: @APJohns  removed added CSS

File: src/app/table/table-footer/display-density/display-density.component.scss (hunk 1 above)
Comment (devendra940) [reply src/app/table/table-footer/display-density/display-density.component.scss]: @APJohns  removed  added CSS

File: src/app/table/table-footer/display-density/display-density.component.html [hunk 7]
```diff
@@ -0,0 +1,9 @@
+<div class="table-display-density">
+    <label class="display-density-label" for="displayDensity">Display Density:</label>
+    <select id="display-density" name="displayDensity" class="display-density-dropdown" (change)="setDisplayDensity($event)"
```
[the remark this answers is not in the document]
Comment (devendra940) [reply src/app/table/table-footer/display-density/display-density.component.html]: @APJohns  updated with displayDensity ID selector

**Comment by devendra940** [comment]
Since I have fixed the review comments and pulled in the latest changes from master to feature/display-density branch, I am reopening this Pull request

File: src/app/table/table.component.ts [hunk 8]
```diff
@@ -152,4 +159,17 @@ export class TableComponent implements OnInit, OnChanges {
     this.endIndex = this.startIndex + this.numberOfRows;
     this.tableData = this.filteredData.slice(this.startIndex, this.endIndex);
   }
+
+    // --------------- DisplayDensity ---------------
+
+    setDisplayDensity(densityName) {
+    if (densityName === undefined) { this.densityClassName = null; }
+    const selectedDensityValue = densityName;
```
Comment (APJohns) [inline src/app/table/table.component.ts]: We can just use the densityName variable, we don't need to create a new variable here

File: src/app/table/table.component.ts [hunk 9]
```diff
@@ -152,4 +159,17 @@ export class TableComponent implements OnInit, OnChanges {
...
+    if (selectedDensityValue === 'Comfortable') {
+      this.densityClassName = null;
+    } else {
+      this.densityClassName = 'table-compact';
+    }
+    localStorage.setItem('selectedView', selectedDensityValue);
```
Comment (APJohns) [inline src/app/table/table.component.ts]: Rename selectedView to selectedDensity. Keeps the name we are using to represent the display density consistent.

File: src/app/table/table-footer/display-density/display-density.component.ts [hunk 10]
```diff
@@ -0,0 +1,32 @@
...
+  displayDensity = 'Comfortable';
+  displayDensityOptions = [
+    { id: 1, selectedValue: 'Comfortable' },
+    { id: 2, selectedValue: 'Compact' }
+  ];
+
+  constructor() { }
+
+  ngOnInit() {
+    this.displayDensity = ( !localStorage.getItem('selectedView') ) ? this.displayDensity : localStorage.getItem('selectedView');
```
[the remark this answers is not in the document]
Comment (devendra940) [reply src/app/table/table-footer/display-density/display-density.component.ts]: @APJohns Fixed

File: src/app/table/table.component.ts (hunk 8 above)
Comment (devendra940) [reply src/app/table/table.component.ts]: @APJohns Fixed

File: src/app/table/table.component.ts (hunk 9 above)
Comment (devendra940) [reply src/app/table/table.component.ts]: @APJohns Fixed

**Comment by devendra940** [comment]
@APJohns 
1) Rename selectedView to selectedDensity 
2) Use the densityName variable instated of creating a new one
3)  Resolve onload nothing to be selected in dropdown now default selected should be comfortable

**Pull request merged**